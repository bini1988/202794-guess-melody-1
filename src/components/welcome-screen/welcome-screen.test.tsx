import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import WelcomeScreen from "./welcome-screen";

expect.addSnapshotSerializer(serializer);

it(`WelcomeScreen correctly renders default markup`, () => {
  const tree = renderer
    .create(
        <WelcomeScreen
          maxTime={5}
          maxMistakes={3}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
