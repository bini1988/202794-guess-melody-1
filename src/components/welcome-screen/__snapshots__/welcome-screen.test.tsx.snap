// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen correctly renders default markup 1`] = `
.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: relative;
  box-sizing: border-box;
  padding: 210px 0px 130px;
  width: 100%;
  height: 100%;
}

.emotion-0 {
  margin: 25px 0px;
}

.emotion-3 {
  padding: 0;
  margin: 75px 0px 35px;
  margin-left: 20px;
  width: 100px;
  height: 140px;
  background: transparent;
  border: none;
  outline: none;
  -webkit-transition: -webkit-transform ease-in-out 300ms;
  -webkit-transition: transform ease-in-out 300ms;
  transition: transform ease-in-out 300ms;
  cursor: pointer;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation-duration: 1.0s;
  animation-duration: 1.0s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

.emotion-3:not(:disabled):hover,
.emotion-3:not(:disabled):focus {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.emotion-3:not(:disabled):active {
  -webkit-transform: scale(1.04);
  -ms-transform: scale(1.04);
  transform: scale(1.04);
}

.emotion-3 > svg {
  fill: #ff9749;
  -webkit-filter: drop-shadow(5px 5px 5px rgba(0,0,0,0.4));
  filter: drop-shadow(5px 5px 5px rgba(0,0,0,0.4));
}

.emotion-2 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: 0;
  padding: 0;
  white-space: nowrap;
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  overflow: hidden;
}

.emotion-5 {
  margin: 0;
  font-style: italic;
  font-weight: 300;
  font-size: 33px;
  color: #230d1a;
  margin-top: 20px;
}

.emotion-7 {
  margin: 10px 0px;
}

.emotion-9 {
  margin: 0;
  padding: 0;
  list-style: none;
  text-align: center;
}

<section
  className="emotion-13 emotion-14"
>
  <div
    className="emotion-0 emotion-1"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="emotion-3 emotion-4"
  >
    <svg
      height="140"
      width="100"
    >
      <polygon
        points="0,0 100,70 0,140"
      />
    </svg>
    <span
      className="emotion-2"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="emotion-5 emotion-6"
  >
    Правила игры
  </h2>
  <p
    className="emotion-7 emotion-8"
  >
    Правила просты:
  </p>
  <ul
    className="emotion-9 emotion-10"
  >
    <li>
      За 5 минут нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 3 ошибки.
    </li>
  </ul>
  <p
    className="emotion-7 emotion-8"
  >
    Удачи!
  </p>
</section>
`;
